### DISCUSSION TITLE
Button inside table row click interfering with onRow click and expandable row click

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#47962] [Q&A] [answered]
Hello, I am trying to get my table row to work properly, when I click a button inside a row, it expands the row as well in addition to triggering the button onClick callback. I tried using event.stopPropagation inside the button callback but it will not prevent the row from expanding/closing when hitting the button. How can one prevent the row onClick to run when a button is inside the table?

**Comment by hbatalhaStch** [comment, marked as answer]
Don't know if you still have this issue but I faced this and solved it like this: 

```tsx
render: (_, request) => {
    return (
        <Button
            type='primary'
            size='small'
            onClick={e => {
                e.stopPropagation()
            }}
        >
            See requests
        </Button>
    )
}
```

**Reply by bombillazo** [reply]
Hey! Yes we solved it by wrapping the buttong/element we wanted to click with a div that on click would stopPropagation.

**Reply by hbatalhaStch** [reply]
It will be useful for new comers
